packages/core/src/hooks/use_item_callbacks.js

```diff
@@ -1,14 +1,17 @@
 import { useState } from "react";
 
-const useItemCallbacks = onSelect => {
+const useItemCallbacks = ({ onSelect, markSelectedItem }) => {
   const [currentDepth, setCurrentDepth] = useState(0);
   const [parents, setParents] = useState([]);
+  const [selectedItem, setSelectedItem] = useState({ item: [], leaf: "" });
 
   const onClick = (label, item, hasChild) => {
     if (hasChild) {
       setParents(parents.concat(label));
       setCurrentDepth(currentDepth + 1);
     } else {
+      markSelectedItem &&
+        setSelectedItem({ item, leaf: item[item.length - 1] });
```

Comment:
I think this is a bit of a mixed pattern.
This means that when you have markSelectedItem you might have a. mismatch with the store in the consumer.

In my opinion this component should be controlled first which means that the selectedItem should come from the consumer and passed as a prop.
It's odd to have both setSelectedItem AND onSelect.

Reply:
@liorheber Do you mean that we will get both selectedItem and onSelect in the props?
And all the logic of the hook will be developed by the consumer?

Reply:
Just like in an input when you use it in it's controlled form

If you do something like this - 
```<input onChange={console.log} />```
You will see the change in the console but the value in the component will never change.
Now if you do it this way -
```<input onChange={callback} value={value} />```
If your callback changes the value it will be reflected in the input and if you place something in the value it will reflect in the input regardless of onChange.
Same here - you should accept the value as a prop in tree and display the selection based on that and not hold the value internally and inform the consumer of it - what if the consumer decided the selection is invalid? or wants to update it later on without user selection (like in load).

Reply:
@liorheber Great. thanks.
If I understand correctly, now we wont need markSelectedItem prop anymore, right?